scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -44,10 +48,13 @@ def process_response(self, request, response, spider):
                     # force recalculating the encoding until we make sure the
                     # responsetypes guessing is reliable
                     kwargs['encoding'] = None
+                if self.keep_encoding_header:
+                    kwargs['flags'] = response.flags
+                    kwargs['flags'].append(b'decoded')
                 response = response.replace(**kwargs)
-                if not content_encoding:
-                    del response.headers['Content-Encoding']
-
+                if not self.keep_encoding_header:
```

Comment:
:lipstick:  I’m not sure if it makes the code more readable or less readable, but you could do ``if not self.keep_encoding_header and not content_encoding:``